src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -1060,6 +1050,20 @@ void CouplingSchemeConfiguration::checkIfDataIsExchanged(
                 dataName);
 }
 
+int CouplingSchemeConfiguration::getWaveformUsedOrder(std::string dataName) const
+{
+  for (const precice::impl::PtrParticipant &participant : _participantConfig->getParticipants()) {
+    for (auto &dataContext : participant->readDataContexts()) {
+      if (dataContext.getDataName() == dataName) {
+        int usedOrder = dataContext.getInterpolationOrder();
+        PRECICE_ASSERT(usedOrder >= 0); // ensure that usedOrder was set
+        return usedOrder;
+      }
+    }
+  }
+  PRECICE_ERROR("Name \"{}\" not found. This is probably a bug. Please report it under https://github.com/precice/precice/issues/new/choose.", dataName);
```

Comment:
I'm raising an error here, to make sure that the order that we receive from this function is actually valid.